hooks/mari/publish_textures.py

```diff
@@ -183,23 +182,28 @@ def _validate_udims_to_copy(self, task_settings, item, required_udims, selected_
                 }
             )
 
-            for udim in required_udims:
-                udim_path = publisher.util.get_path_for_frame(latest_published_path, frame_num=udim)
-                if not os.path.exists(udim_path):
-                    self.logger.error(
-                        "UDIM {} not selected and not previously published!".format(udim),
-                        extra={
-                            "action_show_more_info": {
-                                "label": "Show Error",
-                                "tooltip": "Show more info",
-                                "text": "Previous publish path not found on disk:\n{}\nPlease select "
-                                        "UDIM {} to export it from this session.".format(udim_path, udim)
-                            }
+
+            udim_pattern = publisher.util.get_path_for_frame(latest_published_path, "*")
+            udim_files = glob.glob(udim_pattern)
+            available_udims = {int(publisher.util.get_frame_number(path)) for path in udim_files}
+            non_available_udims = required_udims - available_udims
+
```

Comment:
In my case of 19 udim, 4 selected, the previous os.path.exists() method took ~2.8 sec, while glob.glob() took only 1.4 sec. So replacing that.